src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -153,6 +153,24 @@ private static bool IsEndOfLine(SyntaxTrivia trivia)
                 return null;
             }
 
+            private static bool IsEmbeddedStatement(SyntaxNode node)
+            {
+                return node.Parent switch
+                {
+                    DoStatementSyntax p => node == p.Statement,
+                    ElseClauseSyntax p => node == p.Statement,
+                    FixedStatementSyntax p => node == p.Statement,
+                    CommonForEachStatementSyntax p => node == p.Statement,
+                    ForStatementSyntax p => node == p.Statement,
+                    IfStatementSyntax p => node == p.Statement,
+                    LabeledStatementSyntax p => node == p.Statement,
+                    LockStatementSyntax p => node == p.Statement,
+                    UsingStatementSyntax p => node == p.Statement,
+                    WhileStatementSyntax p => node == p.Statement,
+                    _ => false,
+                };
+            }
```

Comment:
Can you remove this and use IsEmbeddedStatementOwner instead?

Reply:
Not sure; I saw IsEmbeddedStatementOwner, but it is defined in a higher-level project, so I would need to add a non-trivial reference then.